### DISCUSSION TITLE
Using Generics in TypeScript

### DISCUSSION THREAD

**Discussion by davicostalf** [vtex/shoreline#730] [General]
Comentário no CSAT de Engenharia em 2023 H1:

> Tipagens as vezes deixam a desejar, em geral, quando envolvem passar genéricos. Por exemplo, a tipagem do admin-ui-form é uma experiência pior do que utilizar o react-hook-forms diretamente.

**Comment by matheusps** [comment]
This looks like something that we should fix on each occurrence. So... should be issue-based, not discussion-based. I encourage everyone to open issues regarding Typescript types.

**Reply by davicostalf** [reply]
@matheusps About admin-ui-form specifically, do you see any issue that we should create?
